src/neo-vm/ExecutionEngine.cs

```diff
@@ -289,7 +305,43 @@ private bool ExecuteInstruction()
                     }
                 case OpCode.THROW:
                     {
-                        return false;
+                        if (!TryPop(out StackItem error)) return false;
+                        FaultState.Exception = new CatcheableException(error);
+                        return true;
+                    }
+                case OpCode.TRY:
+                    {
+                        int catchOffset = instruction.TokenI8;
+                        int finallyOffset = instruction.TokenI8_1;
+                        return ExecuteTry(catchOffset, finallyOffset);
+                    }
+                case OpCode.TRY_L:
+                    {
+                        int catchOffset = instruction.TokenI32;
+                        int finallyOffset = instruction.TokenI32_1;
+                        return ExecuteTry(catchOffset, finallyOffset);
+                    }
+                case OpCode.ENDT:
+                    {
+                        return ExecuteEndTryCatch(TryState.Try);
+                    }
+                case OpCode.ENDC:
+                    {
+                        return ExecuteEndTryCatch(TryState.Catch);
+                    }
+                case OpCode.ENDF:
+                    {
+                        if (CurrentContext.TryStack is null) return false;
+                        if (!CurrentContext.TryStack.TryPop(out TryContext currentTry))
+                            return false;
+
+                        if (FaultState.Rethrow)
+                        {
+                            FaultState.Rethrow = false;
+                            FaultState.HasCatchableInterrupt = true;
+                            return true;
+                        }
+                        break;
```

Comment:
@lightszero  I have removed `EndPointer` of `TryContext`, as we can't insert an end jump after the syscall, which may also throw a catchable exception, like block/tx not found. Therefore,  the instruction logic of ENDF is adjusted, vm will jump to the next instruction of `ENDF`, which means we also need to insert an end jump instruction of each `ENDF` in compiler.

Reply:
endf without throw，goto next
endf with throw，goto catch

if you goto next always，how to deside where to go？

Reply:
I think that‘s a bug.

endf without throw，goto next
endf with throw，goto catch

if you goto next always，how to deside where to go？

Reply:
> endf with throw，goto catch

Why?  I think it will works well. 

```
case 1:
try-endtry
finally-endfinally + end jump

case 2:
try-endtry
catch-endcatch
finally-endfinally + end jump

case3:
try-endtry + end jump
catch-endcatch + end jump
```

Reply:
```
real case 
try
endtry
addr1:
//codeaftertry
catch
endcatch
addr2:
//codeaftercatch
finally
endfinally
jump here? how to decide jmp addr1 or addr2?
```
when  with throw,you should jump addr2 or else you should jump addr2.

Reply:
if you want handle a  error in syscall,why you modify ENDF,i can not got your point.

Reply:
```c#
try
{
   ...blabla
  return 0;//this is code after try
}
catch
{
   return 1;//this is code after catch
}

```
allow code after try and code after catch will very helpful on code like that.
that's why we decide ENDF logic before.that 's a importent design.

Reply:
Thanks, you are right, we need `EndPointer`, I'll reset it.